### DISCUSSION TITLE
MIDI CC Messages for the Performance parameters, please

### DISCUSSION THREAD

**Discussion by BobanSpasic** [probonopd/MiniDexed#327] [Q&A]
Dear developers, please add MIDI CC Messages for the Performance parameters.
Editing the performances from the encoder is just painful...

**Comment by Doug2233** [comment]
If anything this would be my suggestion too especially if there is a facility to associate the particular knob or slider to a parameter.   The Korg nanoKontrol2 is a perfect candidate for this, and what is more I doubt anyone could build something similar for less than the list price.
![A4A2F840-FF90-46DD-A3A0-A665B09B9BCD](https://user-images.githubusercontent.com/80731201/184841011-e2e66b7c-3b57-4f21-a61a-e6690fc9d97b.jpeg)

**Comment by probonopd** [comment]
I was thinking the other way around: While ANY menu item is open in which you can change a parameter, wiggle any knob/button on any attached MIDI controller, and it would "learn" that whatever MIDI CC the controller is sending should be controlling that parameter: https://github.com/probonopd/MiniDexed/issues/63

This way we would not even have to think about a MIDI representation for the Performance parameters...

Wdyt, would that fulfill your requirement?

**Reply by BobanSpasic** [reply]
Sorry, but I think you are going in the wrong way with implementing things.
That CC mapping should be done as linking custom commands to the established fixed parameter controls.

As a first step, you need to define the parameter numbers and type (boolean, integer, char... or everything as integer with max. range 127 because of MIDI standards) for the performance parameters, implement receiving it as a dump, and as a control change, and that custom mapping after that.

Dumps and CCs should be standardized for exchanging files/parameters between users. Mapping should be user/equipment-specific on the top of that.

**Reply by probonopd** [reply]
You are probably right in describing how to implement things correctly "the MIDI way". We'd need a volunteer to do the work, though.

**Comment by Doug2233** [comment]
I agree with this style of configuration, as intuitive as possible.  

Two thoughts,  could there be a context sensitivity be incorporated.  Eg in one context the faders for example might adjust volume/pan of each Tone Generator yet within a TG they adjust parameters.
The second, persistence. Could a configuration be maintained across reboots?

I may be over reaching but the the options are endless and I’d rather have a simple solution than nothing.  The facility to dynamically change parameters would be such a huge leap.

**Reply by probonopd** [reply]
> The second, persistence. Could a configuration be maintained across reboots?

You can already now save a performance as the default and have it maintained across reboots.
